### PR TITLE
Adds __repr__() to DrawingClassifier_beta class

### PR DESCRIPTION
**Pull request by shantanuchhabra**
This PR will help check off the "Drawing Classifier" box in #2606

### CODE REVIEW

File: src/toolkits/drawing_classifier/drawing_classifier.cpp [hunk 1]
```diff
@@ -508,6 +508,10 @@ void drawing_classifier::iterate_training() {
         {"validation_accuracy", average_val_accuracy},
         {"validation_log_loss", average_val_loss},
     });
+  } else {
+    add_or_update_state({
+      {"validation_accuracy", FLEX_UNDEFINED}
```
Comment (srikris) [inline src/toolkits/drawing_classifier/drawing_classifier.cpp]: We should add `validation_log_loss` hear ideally.

Comment (shantanuchhabra) [reply src/toolkits/drawing_classifier/drawing_classifier.cpp]: Done, thanks!

File: src/toolkits/drawing_classifier/drawing_classifier.cpp [hunk 2]
```diff
@@ -506,7 +506,10 @@ void drawing_classifier::iterate_training() {
   if (validation_data_iterator_) {
     add_or_update_state({
         {"validation_accuracy", average_val_accuracy},
-        {"validation_log_loss", average_val_loss},
+    });
+  } else {
+    add_or_update_state({
```
Comment (Jarvi-Izana) [inline src/toolkits/drawing_classifier/drawing_classifier.cpp]: probably we don't want the user to be aware of the validation metrics if we don't have validation data set.

File: src/toolkits/drawing_classifier/drawing_classifier.cpp [hunk 3]
```diff
@@ -599,6 +605,9 @@ void drawing_classifier::train(gl_sframe data, std::string target_column_name,
     }
   }
 
+  state_update["num_examples"] = data.size();
```
Comment (Jarvi-Izana) [inline src/toolkits/drawing_classifier/drawing_classifier.cpp]: `num_of_training_example` might be more clear? To me, it's a bit ambiguous since it doesn't tell me whether it includes the validation set of not.

File: src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py [hunk 4]
```diff
@@ -1180,3 +1185,37 @@ def evaluate(self, dataset, metric='auto'):
...
+            ('Feature column', 'feature'),
+            ('Target column', 'target')
+        ]
+        training_fields = [
+            ('Training Iterations', 'max_iterations'),
+            ('Training Accuracy', 'training_accuracy'),
+            ('Validation Accuracy', 'validation_accuracy'),
+            ('Training Time', 'training_time'),
+            ('Number of Examples', 'num_examples'),
+            ('Final Loss (specific to model)', 'training_log_loss')
```
Comment (shantanuchhabra) [inline src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: On further thought, for backward compatibility reasons, I don't actually think we should change the name of the state variable from `training_loss` to `training_log_loss`. @nickjong @znation @srikris  What do you think?

File: src/toolkits/drawing_classifier/drawing_classifier.cpp (hunk 3 above)
Comment (srikris) [reply src/toolkits/drawing_classifier/drawing_classifier.cpp]: Sadly we have used `num_examples` everywhere :(. We have to change it everywhere if we do it here.

File: src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py (hunk 4 above)
Comment (srikris) [reply src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: We should leave it as `training_loss`. You might change the loss in the future.

File: src/toolkits/drawing_classifier/drawing_classifier.cpp (hunk 2 above)
Comment (srikris) [reply src/toolkits/drawing_classifier/drawing_classifier.cpp]: But it adds it to the API exposed via Python so it's better for the API to be consistent irrespective of the options that are used during the training process.

File: src/python/turicreate/toolkits/_model.py [hunk 5]
```diff
@@ -513,6 +513,9 @@ def summary(self, output=None):
             return self.__class__.__name__
 
 
+    def _list_fields(self):
```
Comment (srikris) [inline src/python/turicreate/toolkits/_model.py]: This seems strange. You are simply calling the same function by itself? Am I missing something here. Isn't this a recursive call?

Comment (shantanuchhabra) [reply src/python/turicreate/toolkits/_model.py]: Yeah, this is an oversight. Taking it out. Will rebase and squash to update the base commit for all other `__repr__` PRs.

File: src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py (hunk 4 above)
Comment (shreyajain17) [reply src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py]: Btw I removed loss in #2662
